realm/realm-library/src/main/cpp/io_realm_internal_OsList.cpp

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2017 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "io_realm_internal_OsList.h"
+
+#include <list.hpp>
+#include <results.hpp>
+
+#include "util.hpp"
+
+using namespace realm;
+
+static void finalize_list(jlong ptr)
+{
+    TR_ENTER_PTR(ptr)
+    delete reinterpret_cast<List*>(ptr);
+}
+
+JNIEXPORT jlong JNICALL Java_io_realm_internal_OsList_nativeGetFinalizerPtr(JNIEnv*, jclass)
+{
+    TR_ENTER()
+    return reinterpret_cast<jlong>(&finalize_list);
+}
+
+JNIEXPORT jlongArray JNICALL Java_io_realm_internal_OsList_nativeCreate(JNIEnv* env, jclass, jlong shared_realm_ptr, jlong row_ptr,
+                                                                        jlong column_index)
+{
+    TR_ENTER_PTR(row_ptr)
+
+    try {
+        auto& row = *reinterpret_cast<realm::Row*>(row_ptr);
+
+        if (!ROW_VALID(env, &row)) {
+            return 0;
+        }
+        if (!ROW_AND_COL_INDEX_AND_TYPE_VALID(env, &row, column_index, type_LinkList)) {
+            return 0;
+        }
+
+        auto& shared_realm = *reinterpret_cast<SharedRealm*>(shared_realm_ptr);
+        auto link_view_ref = LinkViewRef(row.get_linklist(column_index));
+        auto list_ptr = new List(shared_realm, link_view_ref);
+
+        Table* target_table_ptr = &(link_view_ref)->get_target_table();
+        LangBindHelper::bind_table_ptr(target_table_ptr);
```

Comment:
should we call `unbind_table_ptr` in `~` ?

Reply:
the pointer will be used to create a `Table` and unbind will be called when the `Table` gets GCed.